lib/spack/spack/directory_layout.py

```diff
@@ -190,9 +198,6 @@ def hidden_file_paths(self):
     def relative_path_for_spec(self, spec):
         _check_concrete(spec)
 
-        if spec.external:
```

Comment:
Why do we delete these lines?

Reply:
I moved this handling into `path_for_spec`